tests/test-extensions/petstore-extension/src/main/java/org/mule/test/petstore/extension/PetStoreOperations.java

```diff
@@ -134,6 +135,38 @@ public Boolean isSplitterExceptionHandled() {
     return client.getPets(ownerName, config);
   }
 
+  public List<String> getPetsWithParameterGroup(@Connection PetStoreClient client,
+                                                @Config PetStoreConnector config,
+                                                @ParameterGroup(name = "Owner") PetOwner owner) {
+    return getPets(client, config, owner.getName(), owner.getSignature());
+  }
+
+  public List<String> getPetsWithParameterGroupShowDsl(@Connection PetStoreClient client,
+                                                       @Config PetStoreConnector config,
+                                                       @ParameterGroup(name = "Owner", showInDsl = true) PetOwner owner) {
+    if (owner.getAddress() != null) {
+      IOUtils.toString(owner.getAddress().getValue());
+    }
+    return getPets(client, config, owner.getName(), owner.getSignature());
+  }
+
+  public List<String> getPetsWithParameterGroupShowDslParameterWithAlias(@Connection PetStoreClient client,
```

Comment:
you can join this operation with the one above to simplify and just pass null parameters when necessary, no?

Reply:
makes sense, changed